app/main.py

```diff
@@ -2,5 +2,18 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache_storage = {}
+
+    def wrapper(*args, **kwargs) -> Callable:
+
+        key = (args, frozenset(kwargs.items()))
```

Comment:
Using `frozenset` to convert `kwargs.items()` is a good approach to ensure the key is hashable. However, this will not work if the values in `kwargs` are mutable or not hashable themselves. Ensure that all values in `kwargs` are hashable.